client/core/resource_pool.go

```diff
@@ -0,0 +1,45 @@
+package core
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+var (
+	// ErrComputeCapacityReached is returned by the computational resources pool if the computational effort required
+	// by a transaction is higher than what's left in the block.
+	ErrComputeCapacityReached = errors.New("compute capacity reached")
+)
+
+// ComputationalResourcePool tracks the computational resource available in compute units during
+// the execution of the transactions in a block.
+type ComputationalResourcePool uint64
+
+// AddResource makes computational resource available for execution.
+func (pool *ComputationalResourcePool) AddResource(units uint64) *ComputationalResourcePool {
+	if uint64(*pool) > math.MaxUint64-units {
```

Comment:
Will it better to implemeny Add and Equal methods?

Reply:
just left this part of the code as is